src/lm/mikolov-rnnlm-lib.cc

```diff
@@ -762,24 +762,27 @@ void CRnnLM::restoreNet() {   // will read whole network structure
     goToDelimiter(':', fi);    // direct conenctions
     long long aa;
     for (aa = 0; aa < direct_size; aa++) {
-      fscanf(fi, "%lf", &d);
+      unused_size = fscanf(fi, "%lf", &d);
       syn_d[aa] = d;
     }
   }
   if (filetype == BINARY) {
     long long aa;
     for (aa = 0; aa < direct_size; aa++) {
-      fread(&fl, 4, 1, fi);
+      unused_size = fread(&fl, 4, 1, fi);
       syn_d[aa] = fl;
 
-      /*fread(&si, 2, 1, fi);
+      /*unused_size = fread(&si, 2, 1, fi);
         fl = si/(float)(4*256);
         syn_d[aa] = fl;*/
     }
   }
 
   saveWeights();
 
+  // idiom to "use" an unused variable
+  (void) unused_size;
```

Comment:
you might be able to achieve the same thing by declaration
volatile in unused;
but your solution might be better (could get optimized better), however at the cost that you have remember to "use" the unused variable in each path through the code.